bindings/profilers/wall.cc

```diff
@@ -286,6 +287,11 @@ class SignalHandler {
 #endif
 }  // namespace
 
+static int64_t midpoint(int64_t x, int64_t y) {
+  // TODO: remove when we're on C++20 as it has a built-in midpoint
+  return ((x ^ y) >> 1) + (x & y);
```

Comment:
Nice !
Not sure if right shift is guaranteed to keep the sign bit.

Reply:
I tried this small program:
```
#include <cstdint>
#include <limits>
#include <iostream>

static int64_t midpoint(int64_t x, int64_t y) {
  return ((x ^ y) >> 1) + (x & y);
}

static void test(int64_t x, int64_t y) {
    int64_t mid = midpoint(x, y);
    std::cout << x << ' '<< mid << ' ' << y << '\n';
}

int main(int argc, char** argv) {
    test(std::numeric_limits<int64_t>::min(), std::numeric_limits<int64_t>::max());
    test(std::numeric_limits<int64_t>::min(), -1);
    test(1, std::numeric_limits<int64_t>::max());
}
```
and it does print:
```
-9223372036854775808 -1 9223372036854775807
-9223372036854775808 -4611686018427387905 -1
1 4611686018427387904 9223372036854775807
```
so… seems okay.

Reply:
From https://en.cppreference.com/w/cpp/language/operator_arithmetic:
For negative a, the value of a >> b is implementation-defined (in most implementations, this performs arithmetic right shift, so that the result remains negative).
I guess we need to check that it is the case for the platforms we target